portal-impl/src/com/liferay/portal/servlet/filters/secure/BaseAuthFilter.java

```diff
@@ -108,10 +108,13 @@ protected HttpServletRequest basicAuth(
 			}
 
 			if (userId > 0) {
+				user = UserLocalServiceUtil.getUser(userId);
+
 				httpServletRequest = setCredentials(
 					httpServletRequest, httpSession,
-					UserLocalServiceUtil.getUser(userId),
-					HttpServletRequest.BASIC_AUTH);
+					user, HttpServletRequest.BASIC_AUTH);
+
+				httpSession.setAttribute("DIGEST", user.getDigest());
```

Comment:
@nikolettabuza this seems a bit strange. BASIC auth does not use the digest and it is possible that a user will not have a digest. It is only available when they generate a token via Account Settings > Password